### ISSUE TITLE
Cmd.exe closes immediately after running

### ISSUE THREAD

**Issue by kinesivan** [foreshadow/atom-python-run#5]
I'm running Python 3.5, upon pressing F5 or F6, a command prompt pops up but immediately closes. I tried halting the process with a basic `k = input("Press any key to close")` but that didn't seem to work.

**Comment by foreshadow** [comment]
@kinesivan   
I mainly work and test with python 2.7, but I think 3.5 works similarly. As the cpp file in src directory given, I use `system` call to launch python.exe, which will not terminate the program no matter the compilation succeeds or not. That means it will always pause before it closes. Anyway, the latest version fixes some potential bugs, you can try it first. If it doesn't work, I would like to know whether there is a runtime error message or dialog when cmd closes. If not, I need to know your system edition(win7, 8, 10, etc) or more environment detail, because it works fine in my Win7x64 with native cmd and C runtime library, and I suppose many others' work fine too.

**Comment by mattivityroom** [comment]
Hey @foreshadow 
I get a same problem, when i running my python in atom, suddenly it close after open.
I'm using python 2.7.

**Comment by rwjb** [comment]
W10, both 2.7 and 3.4 will exit immediately when tested with an infinite loop.

**Comment by Frezzu** [comment]
I had the same problem but i looked to the code and change some part of it. I also add one dll file to folder bin. So for me problem appeared cause of missing dll file and not using absolute path file for executed python file.

**Missing dll:**
> libgcc_s_dw2-1.dll

**Changes in atom-python-run/bin/cp.exe:**
_Before:_
> char cmd[4096];
>  strcat(cmd, "title ");

_After:_
> char cmd[4096] = "title ";

**Changes in atom-python-run/lib/atom-python-run.js:**
_Before:_
>     var child = child_process.spawn("cmd", [
>         "/c", "start", __dirname + "/../bin/cp.exe", "python.exe", info.base
>     ], { 
>         detach: true 
>     });

_After:_
>     var child = child_process.spawn("cmd", [
>         "/c", "start",  __dirname + "/../bin/cp.exe", "python.exe", file
>     ], { 
>         detach: true 
>     });


I couldn't add files to this post so i created a repository for it. If the creator @foreshadow want to delete this repository just say it to me :)
**Repo(changed files):** https://github.com/Frezzu/atom-python-run-fix

**Comment by foreshadow** [comment]
Thanks to debug messages given and @Frezzu 's solution.
So this bug is mainly because cp.exe requires libgcc which some users don't have.
(When I used g++ to build cp.exe before, I originally thought that it only requires C runtime environment windows has.)

@Frezzu If it is convienient for you, you can fork this repo, fix it as you have, then make a pull request. I am looking forward to it.
(Although it is not a good idea to call `cmd start` to start c++ program in node.js, your solution is a good way to fix it for now.)

**Comment by Frezzu** [comment]
I made the pull. :)

**Comment by TigersWay** [comment]
Hi,
Still with that same trouble: the console window opens and closes instantly (<input> does not help)!
Notifications seem ok
bin\cp.exe can be found and run
Under Win7 | Atom in 'portable mode'

Thanks

**Comment by foreshadow** [comment]
@TigersWay
Hi, I am not familiar with portable mode. However, I want to know what `cp.exe` shows if it run individually in your environment.  The following text is what is expected to be shown:
```
Process returned 0 (0x0)   execution time : 0.123 s\n
Press any key to continue...
```

**Comment by subash774** [comment]
same problem, cmd.exe just flashes, the code isn't run.

**Comment by foreshadow** [comment]
@subash774 
What will happen if you run bin/cp.exe using console command (in which way you might see the output)?

**Comment by subash774** [comment]
Yes when I run cp.exe i see the output (execution time can be seen) but when run from atom using f5, the window just flashes and disappears

**Comment by subash774** [comment]
**Update:** 
After uninstalling atom and python completely, cp.exe still works and I can see the output but when I press F5 i get an error that says "Windows cannot find python.exe E:\...... Make sure you've typed the name correctly, then try again". And yep, python is in environment variables.

**Comment by foreshadow** [comment]
It is pretty wierd if you can execute `python -V` in your console, check it.
In my view, there is no difference between calling `python` from console and using WinApi `system`.
If you are interested, you can have a look at [the source code of cp.cpp](https://github.com/foreshadow/atom-python-run/blob/master/src/cp.cpp).
And [line 74 in atom-python-run.js](https://github.com/foreshadow/atom-python-run/blob/master/lib/atom-python-run.js#L74) might be useful.
After all, it works fine in my Win7 with `PATH` set up correctly.
If anyone has the same problem, please comment below.

**Comment by TigersWay** [comment]
I can confirm last @subash774 comment. 
After updating, I only get this dialog:
![untitled](https://cloud.githubusercontent.com/assets/114064/21959022/a08441d4-daed-11e6-9592-561e134fa81f.png)

of course the same command from any windows/dos console works fine.

**Comment by subash774** [comment]
Thanks @TigersWay  for the confirmation, so we know that this isn't just a bug in 1 computer. @foreshadow ...rollback? or lets compare code with previous versions? that way we can have a look at what changed and go about there?

**Comment by foreshadow** [comment]
I was on my way home a few hours ago.
I think I probably know the reason now. 
It is about the new feature in this version. It seems that I must split the command into different items in array rather than split with space in one string. And I ignored double quotes in the command which will probably make an error when the path contains spaces.
See [line 54 on the left and line 74-76 on the right](https://github.com/foreshadow/atom-python-run/commit/5050cca82f8c8771b23afc30fe9dcd0dfd565522#diff-1ee16f340e164db339b0333a9690f174L54) .
You can help to fix it, or I will fix it in one day.

**Comment by foreshadow** [comment]
By the way, it really works fine in both my computers, and I still don't know why it works differently in different environments .

**Comment by TigersWay** [comment]
Need to add a little something, which is probably not coming from the plugin but the path of the python file - if under junction - is "wrong".
Let me explain :-) Most of files I'm working on are on a long path through dropbox and client version. I so use NTFS junction to make paths shorter, and I work there. But the earlier message "Windows cannot find...." is always with the long/real path.

**Comment by subash774** [comment]
@foreshadow , is there any way i can install previous versions? I dont wanna be messing around with your code as it works fine for others. If i could use previous versions that'd do. thanks

**Comment by foreshadow** [comment]
@subash774 , I published v0.7.2, try it first. 

If it still doesn't work, you can install previous versions as follows:
1. change directory to packages folder of atom
```
cd C:/Users/<your-user-name>/.atom/packages
```
2. delete the package manually or by this command (be careful of the rd /S command)
```
rd atom-python-run /S
```
3. install a previous version using `git clone --branch`, I suppose you want v0.6.2.
```
git clone git@github.com:foreshadow/atom-python-run.git --branch v0.6.2
```
or you can download the zip [at this page](https://github.com/foreshadow/atom-python-run/tree/v0.6.2) and unzip it into `~/.atom/packages`

**Comment by subash774** [comment]
yep v0.7.2 still doesn't work, cloned to your old repo and what a surprise haha flashing problem is back, I'm trying older versions now, will let you know when it works

**Comment by ngqth** [comment]
Hi, i had the exact same problem as describe above, my cmd close instantly after press F5. I found a solution, at least for my case. 

The problem being, i install the atom-python-run in my user folder which is named: "xxx xxx xxx" (with the space in-between). Then i removed all the space from my user folder name, now it looks like this: "xxxxxxx" (without the space). Ta-da it works again. Aint know it will apply to every one or not, but i solved my problem. Hope that helps.

**Comment by subash774** [comment]
I use microsoft account which means my name will have a space in it, IDK how to fix that.

**Comment by lone17** [comment]
Having the same problem in win10 + python 3.6.1
cp.exe works fine since it shows 
```
Process returned 0 (0x0)   execution time : 0.043 s
Press any key to continue . . .
```

What can I do to fix this ?

**Comment by subash774** [comment]
I'm assuming you're signed in with Microsoft account? And you've a space in your profile name? That causes it to crash. There shouldn't be a space in the profile for this to run.

**Comment by jalynxing** [comment]
我也出现了同样的错误。我将setting路径中的{file}外的双引号去掉后，再按F5能够运行一部分，仍然是闪退的结果。

**Comment by foreshadow** [comment]
Spaces and special characters in path has been resolved in v0.8.1.
And we are using a completely new cp(.py) now, which has a better compatibility, and it it cross-platform.
If there is still problems, you can new a new issue with `cp.log`'s information.
